src/auth/strategies/github.strategy.ts

```diff
@@ -0,0 +1,46 @@
+import { Inject, Injectable } from "@nestjs/common";
+import { PassportStrategy } from "@nestjs/passport";
+import { Strategy } from "passport-github2";
+import { GithubProfile } from "@/global/types/auth.types";
+import { IAuthProvider } from "../../global/interfaces/oauth.interface";
+import { OAuthConfig } from "../../config/Oauth/oauthConfig.interface";
+import { InternalServerErrorException } from "@nestjs/common";
+
+@Injectable()
+export class GithubStrategy extends PassportStrategy(Strategy, "github") {
+    constructor(
+        @Inject("GITHUB_OAUTH")
+        private readonly githubAuthService: IAuthProvider,
+        @Inject("OAuth-Config") private oAuthConfig: OAuthConfig,
+    ) {
+        const { clientId, clientSecret, callbackUrl } = oAuthConfig.github;
+        super({
+            clientID: clientId,
+            clientSecret: clientSecret,
+            callbackURL: callbackUrl,
+            scope: ["identify", "user:email"],
+        });
+    }
+
+    async validate(
+        accessToken: string,
+        refreshToken: string,
+        profile: GithubProfile,
+    ): Promise<any> {
+        const { username, id, photos, emails } = profile;
+        const avatar = photos && photos.length > 0 ? photos[0].value : null;
```

Comment:
Might be related to the comment above about the types packages - for readability I think we should explicitly declare what the type of avatar is